syntax = "proto3";

package ticketing.events;

// 1. Ticketing → Reservation: Reserve seats
// Note: config (rows, cols, price) is now fetched by Reservation Service from Kvrocks directly
message BookingCreatedDomainEvent {
  string booking_id = 1;
  int32 buyer_id = 2;
  int32 event_id = 3;
  int32 total_price = 4;
  string section = 5;
  int32 subsection = 6;
  int32 quantity = 7;
  string seat_selection_mode = 8;
  repeated string seat_positions = 9;
  string status = 10;
  // Field 11 reserved (previously config)
  // OpenTelemetry W3C Trace Context
  string traceparent = 12;
  string tracestate = 13;
  // Message type for consumer routing (always "BookingCreatedDomainEvent")
  string message_type = 14;
}

// 2. Ticketing → Reservation: Release seats (minimal - consumer fetches details from DB)
message BookingCancelledEvent {
  string booking_id = 1;
  int32 event_id = 2;
  string section = 3;       // for partition calculation
  int32 subsection = 4;     // for partition calculation
  // OpenTelemetry W3C Trace Context
  string traceparent = 5;
  string tracestate = 6;
  // Message type for consumer routing (always "BookingCancelledEvent")
  string message_type = 7;
}
